runtime/cudaq/platform/default/nvidia.config

```
@@ -1,19 +1,10 @@
 msg=""
-if [ -x "$(command -v nvidia-smi)" ]; then
-	# Make sure nvidia-smi works.
-	nvidia-smi -L | grep 'Failed\|Error\|error\|failed' >/dev/null 2>&1
-	if [ $? != 0 ]; then
-		ngpus=$(nvidia-smi -L | wc -l)
-		if [ -f "${install_dir}/lib/libnvqir-custatevec-fp32.so" ] && [ $((ngpus > 0)) != 0 ]; then
-			NVQIR_SIMULATION_BACKEND="custatevec-fp32"
-		else
-		    msg="libnvqir-custatevec-fp32 is not installed, or there are no NVIDIA GPUs."
-		fi
-	else
-	  msg="nvidia-smi failed with \"$(nvidia-smi -L)\""
-	fi
-else 
-    msg="nvidia-smi command not found."
+
+gpu_found=$(query_gpu)
+if [ "$gpu_found" = true ] && [ -f "${install_dir}/lib/libnvqir-custatevec-fp32.so" ]; then
```

```suggestion
if $(gpu_found) && [ -f "${install_dir}/lib/libnvqir-custatevec-fp32.so" ]; then
```
A little simplification assuming the variable `gpu_found` is set to either `true` or `false` precisely.

---

Done.